pxteditor/monaco-fields/field_sprite.ts

```diff
@@ -0,0 +1,89 @@
+/// <reference path="./monacoFieldEditor.ts" />
+
+namespace pxt.editor {
+    const fieldEditorId = "image-editor";
+
+    export class MonacoSpriteEditor implements MonacoFieldEditor {
+        private resolver: (edit: TextEdit) => void;
+        private rejecter: (err?: any) => void;
+
+        protected editor: pxtsprite.SpriteEditor;
+        protected editrange: monaco.Range;
+
+        getId() {
+            return fieldEditorId;
+        }
+
+        showEditorAsync(editrange: monaco.Range, host: MonacoFieldEditorHost): Promise<TextEdit> {
```

Comment:
It would be good to see us converge the Blockly and Monaco field editors into one interface so we can implement field editors irrespective of where they show up. 

Right now with this current implementation it seems you only allowing field editors to write (set) their value when they close. Here are some examples of field editors that write their value a number of times: 
- note picker
- slider
- turn ratio

Reply:
The current API was designed to be transactional to prevent code from moving around while the field editor is open. I'll mark this as something to revisit after the holidays (when everyone is back), but we'll merge as-is and tweak it once we have some feedback